### PR TITLE
Support runs with heaptrack profiler

### PR DESCRIPTION
**Pull request by squadgazzz**
This PR enables running any service with the `heaptrack` memory profiler. Initially, the plan was to install the necessary components for `heaptrack` only when the `heaptrack` argument is specified. However, the only method I've found involves incorporating the installation script in a bash script that executes every time the image restarts, which isn't good. As a result, the final image is a bit heavier now(1.93GB vs 1.22GB).

There is an option to attach the profiler to an already running process without changing the dockerfile, but this consumes more resources and makes it impossible to collect any results since the API latency increases drastically, which leads to frequent pod restarts. Incorporating the profiler into the image and running it from the beginning has less resource impact.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -52,12 +52,22 @@ ENTRYPOINT [ "solvers" ]
 
 # Extract Binary
 FROM intermediate
+RUN apt-get update && \
```
Comment (MartinquaXD) [inline Dockerfile:55]: How much of a size difference does this make?
AFAICS this should add the entire repo, dependencies and build artifacts to the image.
Also what't the impact on build times for the image?

Comment (squadgazzz) [reply Dockerfile:55]: The build time is the same: https://github.com/cowprotocol/services/actions/runs/9778536743 vs https://github.com/cowprotocol/services/actions/runs/9773882766
The image size is 1.93GB vs 1.22GB.

Comment (MartinquaXD) [reply Dockerfile:55]: Okay not too bad. Does the `heaptrack` also have a `make clean` command to remove the intermediate build artifacts?

Comment (squadgazzz) [reply Dockerfile:55]: Added a clean-up logic.

**Review by MartinquaXD — APPROVED** [review]

**Review by Mateo-mro — APPROVED** [review]

File: entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+# Default to "unknown" if POD_NAME is not set
+pod_name=${POD_NAME:-unknown}
+
+# Check if heaptrack is enabled as the first argument
+if [ "$1" = "heaptrack" ]; then
+    # Remove 'heaptrack' from the arguments
+    shift
+    # Execute the remaining command with heaptrack
+    exec heaptrack -o "/tmp/heaptrack/heaptrack.$pod_name.$(shuf -i 1-99999 -n 1).gz" "$@"
```
Comment (fleupold) [inline entrypoint.sh:11]: Using current timestamp could be useful instead of a random id

File: entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+
+# Default to "unknown" if POD_NAME is not set
+pod_name=${POD_NAME:-unknown}
```
Comment (fleupold) [inline entrypoint.sh:4]: Why don't you use the result of `hostname` here (which is the pod name on k8s machines)?

**Review by fleupold — APPROVED** [review]

File: entrypoint.sh (hunk 3 above)
Comment (squadgazzz) [reply entrypoint.sh]: Ah, true, thanks!

**Review by sunce86 — APPROVED** [review]

**Pull request merged**